sentry/src/transports/ratelimit.rs

```diff
@@ -0,0 +1,141 @@
+use httpdate::parse_http_date;
+use std::time::{Duration, SystemTime};
+
+// TODO: maybe move this someplace where we can filter an `Envelope`s items.
+
+/// A Utility that helps with rate limiting sentry requests.
+#[derive(Debug, Default)]
+pub struct RateLimiter {
+    global: Option<SystemTime>,
+    error: Option<SystemTime>,
+    session: Option<SystemTime>,
+    transaction: Option<SystemTime>,
+}
+
+impl RateLimiter {
+    /// Create a new RateLimiter.
+    pub fn new() -> Self {
+        Self::default()
+    }
+
+    /// Updates the RateLimiter with information from a `Retry-After` header.
+    pub fn update_from_retry_after(&mut self, header: &str) {
+        let new_time = if let Ok(value) = header.parse::<f64>() {
+            Some(SystemTime::now() + Duration::from_secs(value.ceil() as u64))
```

Comment:
Why is rate limiting done with SystemTime rather than Instant?

Reply:
Not quite sure actually. I just copied the old code, without thinking too much about it.